core/cli/src/config.ts

```diff
@@ -159,8 +159,8 @@ export function validateConfig(config: ValidPluginsConfig, logger: Logger): Vali
   const validConfig = config as ValidConfig
 
   const hookTaskConflicts = findConflicts(Object.values(config.hookTasks))
-  const hookConflicts = findConflicts(Object.values(config.hooks))
-  const taskConflicts = findConflicts(Object.values(config.tasks))
+  const hookConflicts = findConflictingEntries(config.hooks)
+  const taskConflicts = findConflictingEntries(config.tasks)
   const optionConflicts = findConflicts(Object.values(config.options))
```

Comment:
**suggestion(non-blocking):** would be great to see some tests for this logic in the future.